cmd/build.go

```diff
@@ -93,3 +120,72 @@ func Build(ctx context.Context) *cobra.Command {
 	cmd.Flags().StringArrayVar(&options.Secrets, "secret", nil, "secret files exposed to the build. Format: id=mysecret,src=/local/secret")
 	return cmd
 }
+
+func buildImageWithOpts(options build.BuildOptions, path string) error {
+
+	if err := utils.CheckIfDirectory(path); err != nil {
+		return fmt.Errorf("invalid build context: %s", err.Error())
+	}
+	options.Path = path
+
+	if options.File == "" {
+		options.File = filepath.Join(path, "Dockerfile")
+	}
+
+	if err := utils.CheckIfRegularFile(options.File); err != nil {
+		return fmt.Errorf("invalid Dockerfile: %s", err.Error())
+	}
+
+	if okteto.Context().Builder == "" {
+		log.Information("Building your image using your local docker daemon")
+	} else {
+		log.Information("Running your build in %s...", okteto.Context().Builder)
+	}
+
+	ctx := context.Background()
+	if err := build.Run(ctx, options); err != nil {
+		analytics.TrackBuild(okteto.Context().Builder, false)
+		return err
+	}
+
+	if options.Tag == "" {
+		log.Success("Build succeeded")
+		log.Information("Your image won't be pushed. To push your image specify the flag '-t'.")
+	} else {
+		log.Success(fmt.Sprintf("Image '%s' successfully pushed", options.Tag))
+	}
+
+	analytics.TrackBuild(okteto.Context().Builder, true)
+	return nil
+}
+
+func buildFromManifest(name string, b *model.BuildInfo) error {
+	if b.Image == "" {
+		b.Image = setOktetoImageTag(name)
+	}
+
+	opts := build.BuildOptions{
+		BuildArgs:  model.SerializeBuildArgs(b.Args),
+		CacheFrom:  b.CacheFrom,
+		Target:     b.Target,
+		Path:       b.Context,
+		Tag:        b.Image,
+		OutputMode: "tty",
```

Comment:
@jLopezbarb what's the envvar to set this in the pipelines? we shouldn't hardcode `tty`